module/canvas/ability-template.mjs

```diff
@@ -223,7 +236,27 @@ export default class AbilityTemplate extends MeasuredTemplate {
     const now = Date.now(); // Apply a 20ms throttle
     if ( now - this.#moveTime <= 20 ) return;
     const center = event.data.getLocalPosition(this.layer);
-    this.document.updateSource(canvas.templates.getSnappedPoint(center));
+    const updates = canvas.templates.getSnappedPoint(center);
+
+    // Adjust template size to take hovered token into account if `adjustedSize` is set
+    const baseDistance = this.document.flags.dnd5e?.dimensions?.size;
+    if ( this.document.flags.dnd5e?.dimensions?.adjustedSize && baseDistance ) {
+      const rectangle = new PIXI.Rectangle(center.x, center.y, 1, 1);
+      const hoveredToken = canvas.tokens.placeables
+        .find(t => t.visible && !t.document.isSecret && t._overlapsSelection(rectangle));
```

Comment:
We should use the quad tree rather than iterating over all tokens:

```suggestion
      const hoveredToken = canvas.tokens.quadtree.getObjects(rectangle, { collisionTest: t => t.visible && !t.document.isSecret }).first();
```